### PR TITLE
[WIP] Handle block params in DuplicateMethodCall

### PR DESCRIPTION
**Pull request by mvz**
#1231.

For now, this just show what I think is the desired behavior. @troessner do you agree that these specs reflect how this should work?

### CODE REVIEW

File: spec/reek/smell_detectors/duplicate_method_call_spec.rb [hunk 1]
```diff
@@ -126,6 +126,40 @@ def alfa
...
+          charlie.each { |delta| delta.echo }
+        end
+      EOS
+      expect(src).not_to reek_of(:DuplicateMethodCall)
+    end
+
+    it 'does not report calls inside lambda expressions' do
+      src = <<-EOS
+        def alfa
+          bravo = ->(delta) { delta.echo.foxtrot }
```
Comment (troessner) [inline spec/reek/smell_detectors/duplicate_method_call_spec.rb]: Why do you need the "foxtrot" at the end? We can drop that, can't we?

**Comment by troessner** [comment]
Same as with #1252 - I agree this is the way to go.

File: spec/reek/smell_detectors/duplicate_method_call_spec.rb (hunk 1 above)
Comment (mvz) [reply spec/reek/smell_detectors/duplicate_method_call_spec.rb]: Probably :-)

**Comment by troessner** [comment]
@mvz this PR seems stale, wdyt about closing it for now?

**Comment by mvz** [comment]
@troessner Agreed. Closing.

**Pull request closed without merging**